File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    ans = 1
    dic = dict()
    
    for c in clothes:
        # dic의 key가 아니면 일단 추가하고, key이면 value값 +1 해주기
        if c[1] not in dic.keys():
            dic[c[1]] = 1
        else:
            dic[c[1]] += 1

    for d in dic.values():
        ans += d * ans
        
    return ans - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let ans = 1
    
    let dict = {}
    
    for (c of clothes) {
        if (!Object.keys(dict).includes(c[1])) {
            dict[c[1]] = 1
        } else {
            dict[c[1]]++
        }
    }
    
    for (v of Object.values(dict)) {
        ans += ans * v
    }
    

    return ans-1;
}